### ISSUE TITLE
Bug: typescript-estree should allow `using` declaration in ambient context

### ISSUE THREAD

**Issue by JLHwung** [typescript-eslint/typescript-eslint#11244]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.3&sourceType=script&fileType=.ts&code=CYUwxgNghgTiAEA7KBbEBnADlMCBuUEAlsAPrJpY4ikEzwDeAUPPAK7pGIDm8AIgG4mAXyZA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=true

### Repro Code

```TypeScript
declare namespace valid_namespace_var {
  using D;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
};
```

### tsconfig

```jsonc
See REPL
```

### Expected Result

Pass. The TS Playground is happy about that: https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEA7KBbEBnADlMCBuUEAlsAPrJpY4ikEzwDeAUPPAK7pGIDm8AIgG4mAXyZA

### Actual Result

Error is thrown from https://github.com/typescript-eslint/typescript-eslint/blob/4ad8a0713376e5b88c56997312861a2c8cb049ab/packages/typescript-estree/src/convert.ts#L1247

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `8.32.1` |

**Comment by JLHwung** [comment]
> Do you have a real world usecase for this? Cos honestly I'm pretty okay with this failing given its pretty nonsensical code... 

I don't expect there are real world case, this feature just begins to ship very recently.

Given that tsc already accept `using D` in ambient context, I don't think TSESLint should reject valid TS on the parser level. And using declaration in an ambient context is not nonsensical: compared with `const C`, `using D` tells the type compiler that `D` must be a Disposable or nullish value: way more narrowed than `const C`. 

Of course if tsc decides to forbid `using` in ambient context, then we can close this issue, but I don't think this is a decision that we should made.

**Comment by bradzacher** [comment]
Yeah I'm _personally_ okay with us being incorrect here because having to detect the "ambient declaration context" is a pain and not something we do in the parser right now. Without a real-world usecase I don't personally see the motivation to build that logic in -- I don't think that academic correctness is a good enough reason.

cc @typescript-eslint/triage-team thoughts?

**Comment by JLHwung** [comment]
Closing this issue as tsc has disallowed using in ambient context.
